src/test/kotlin/test/ktortemplate/core/utils/healthcheck/TestHealthCheck.kt

```diff
@@ -23,44 +26,79 @@ class TestHealthCheck : KoinTest {
     companion object {
         @Container
         private val dbContainer = PgSQLContainerFactory.newInstance()
+        private val log = LoggerFactory.getLogger(this::class.java)
     }
 
     @Test
     fun `Checking liveness with 200OK`() = testAppWithConfig {
         with(handleRequest(HttpMethod.Get, "/liveness")) {
-            response.status() `should be equal to` io.ktor.http.HttpStatusCode.OK
-            val result: Map<String, String> = JsonSettings.mapper.readValue(response.content!!)
-            result["alive"].toBoolean() `should be equal to` true
+            response.status() `should be equal to` HttpStatusCode.OK
+            val result: Map<String, Boolean> = JsonSettings.mapper.readValue(response.content!!)
+            result["alive"] `should be equal to` true
         }
     }
 
     @Test
     fun `Checking database readiness with 200OK`() = testAppWithConfig {
         with(handleRequest(HttpMethod.Get, "/readiness")) {
-            response.status() `should be equal to` io.ktor.http.HttpStatusCode.OK
-            val result: Map<String, String> = JsonSettings.mapper.readValue(response.content!!)
-            result["database"].toBoolean() `should be equal to` true
+            response.status() `should be equal to` HttpStatusCode.OK
+            val result: Map<String, Boolean> = JsonSettings.mapper.readValue(response.content!!)
+            result["database"] `should be equal to` true
         }
     }
 
     @Test
-    fun `Checking readiness with 500 Internal Server Error`() = testAppWithConfig {
-        with(handleRequest(HttpMethod.Get, "/readiness")) {
-            response.status() `should be equal to` io.ktor.http.HttpStatusCode.OK
-            val result: Map<String, String> = JsonSettings.mapper.readValue(response.content!!)
-            result["database"].toBoolean() `should be equal to` true
+    fun `Checking readiness with 500 Internal Server Error`() {
+        val dbContainer = PgSQLContainerFactory.newInstance()
```

Comment:
I suspect the problem had to do with a timing issue in shutting down the container, so for this test we create a new container that is manually managed